From 4faf68d68c97cfd10947e1152f711acc59f39647 Mon Sep 17 00:00:00 2001
From: Erik Jansson <erikja@axis.com>
Date: Wed, 16 Oct 2019 15:07:48 +0200
Subject: [PATCH] Finding host-local in /usr/libexec

Upstream-status: Inappropriate [embedded specific]
Signed-off-by: <erikja@axis.com>
---
 pkg/agent/config/config.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: import/pkg/agent/config/config.go
===================================================================
--- import.orig/pkg/agent/config/config.go
+++ import/pkg/agent/config/config.go
@@ -445,7 +445,7 @@
 	}
 
 	if !nodeConfig.NoFlannel {
-		hostLocal, err := exec.LookPath("host-local")
+		hostLocal, err := exec.LookPath("/usr/libexec/cni/host-local")
 		if err != nil {
 			return nil, errors.Wrapf(err, "failed to find host-local")
 		}
