From 29865b7e9288a60aca151df323ee89c0952e8158 Mon Sep 17 00:00:00 2001
From: Jianchuan Wang <jianchuan.wang@windriver.com>
Date: Thu, 3 Nov 2016 01:33:57 -0400
Subject: [PATCH] criu: Correct the installation directory for python module
 distributions

Signed-off-by: Jianchuan Wang <jianchuan.wang@windriver.com>
---
 lib/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/Makefile b/lib/Makefile
index f1c0821..c3003a5 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -55,8 +55,8 @@ install: lib-c lib-py ../crit/crit c/criu.pc.in
 	$(Q) mkdir -p $(DESTDIR)$(LIBDIR)/pkgconfig
 	$(Q) sed -e 's,@version@,$(CRIU_VERSION),' -e 's,@libdir@,$(LIBDIR),' -e 's,@includedir@,$(dir $(INCLUDEDIR)),' c/criu.pc.in > c/criu.pc
 	$(Q) install -m 644 c/criu.pc $(DESTDIR)$(LIBDIR)/pkgconfig
-	$(E) "  INSTALL " crit
-	$(Q) python ../scripts/crit-setup.py install --root=$(DESTDIR) --prefix=$(PREFIX) --record $(CRIT_SETUP_FILES)
+	$(E) "  INSTALL " crit 
+	$(Q) python ../scripts/crit-setup.py install --root=$(DESTDIR) --prefix=$(PREFIX) --install-lib=$(PYTHON_SITEPACKAGES_DIR) --record $(CRIT_SETUP_FILES)
 .PHONY: install
 
 uninstall:
-- 
2.8.1

