From b4ad9b2a57d6c736870657c06959ccc44b578f12 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@gmail.com>
Date: Tue, 30 Jun 2020 23:47:08 -0400
Subject: [PATCH] build: use  for cross compiler

Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 a/src/import/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: oci-runtime-tools-0.9.0+gitAUTOINC+0105384f68/src/import/Makefile
===================================================================
--- oci-runtime-tools-0.9.0+gitAUTOINC+0105384f68.orig/src/import/Makefile
+++ oci-runtime-tools-0.9.0+gitAUTOINC+0105384f68/src/import/Makefile
@@ -13,11 +13,11 @@
 all: tool runtimetest validation-executables
 
 tool:
-	go build $(BUILD_FLAGS) -o oci-runtime-tool ./cmd/oci-runtime-tool
+	${GO} build $(BUILD_FLAGS) -o oci-runtime-tool ./cmd/oci-runtime-tool
 
 .PHONY: runtimetest
 runtimetest:
-	go build $(STATIC_BUILD_FLAGS) -o runtimetest ./cmd/runtimetest
+	${GO} build $(STATIC_BUILD_FLAGS) -o runtimetest ./cmd/runtimetest
 
 .PHONY: man
 man:
