From 3c51d942f6da08045351ce61cc7f426fa0855489 Mon Sep 17 00:00:00 2001
From: Chen Qi <Qi.Chen@windriver.com>
Date: Fri, 31 Aug 2018 10:51:36 +0800
Subject: [PATCH] fix autoheader error

Upstream-Status: Pending

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 configure.ac | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/configure.ac b/configure.ac
index 17e436d..16b3e71 100644
--- a/configure.ac
+++ b/configure.ac
@@ -388,7 +388,7 @@ AC_ARG_WITH(iobroker,
 case $IOBROKER_METHOD in
 epoll*)
 	if test "$GLIBC_NEWER_2_4" -eq 1 -a "x$ac_cv_header_sys_epoll_h" = "xyes"; then
-		AC_DEFINE([IOBROKER_USES_EPOLL])
+		AC_DEFINE([IOBROKER_USES_EPOLL], [1], [iobroker uses epoll])
 	else
 		echo "\"epoll\" is not available as an iobroker method."
 		echo "Please use one of the other options."
@@ -397,7 +397,7 @@ epoll*)
 	;;
 poll*)
 	if test "x$ac_cv_header_sys_poll_h" = "xyes" -o "x$ac_cv_header_poll_h" = "xyes"; then
-		AC_DEFINE([IOBROKER_USES_POLL])
+		AC_DEFINE([IOBROKER_USES_POLL], [1], [iobroker uses poll])
 	else
 		echo "\"poll\" is not available as an iobroker method."
 		echo "Please use one of the other options."
@@ -406,7 +406,7 @@ poll*)
 	;;
 select*)
 	if test "x$ac_cv_header_sys_select_h" = "xyes"; then
-		AC_DEFINE([IOBROKER_USES_SELECT])
+		AC_DEFINE([IOBROKER_USES_SELECT], [1], [iobroker uses select])
 	else
 		echo "\"select\" is not available as an iobroker method."
 		echo "Please use one of the other options."
@@ -415,13 +415,13 @@ select*)
 	;;
 none*)
 	if test "$GLIBC_NEWER_2_4" -eq 1 -a "x$ac_cv_header_sys_epoll_h" = "xyes"; then
-		AC_DEFINE([IOBROKER_USES_EPOLL])
+		AC_DEFINE([IOBROKER_USES_EPOLL], [1], [iobroker uses epoll])
 		IOBROKER_METHOD="epoll"
 	elif test "x$ac_cv_header_sys_poll_h" = "xyes" -o "x$ac_cv_header_poll_h" = "xyes"; then
-		AC_DEFINE([IOBROKER_USES_POLL])
+		AC_DEFINE([IOBROKER_USES_POLL], [1], [iobroker uses poll])
 		IOBROKER_METHOD="poll"
 	elif test "x$ac_cv_header_sys_select_h" = "xyes"; then
-		AC_DEFINE([IOBROKER_USES_SELECT])
+		AC_DEFINE([IOBROKER_USES_SELECT], [1], [iobroker uses select])
 		IOBROKER_METHOD="select"
 	else
 		echo "There are no available options for iobroker polling"
-- 
2.7.4

