From ae6f4a1e685394fb7b9539f134b66ca84ddc91c2 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@gmail.com>
Date: Tue, 30 Jun 2020 22:19:56 -0400
Subject: [PATCH] cli: use go cross compiler

Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 git/cli/scripts/build/dynbinary | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: git/cli/scripts/build/dynbinary
===================================================================
--- git.orig/cli/scripts/build/dynbinary
+++ git/cli/scripts/build/dynbinary
@@ -19,6 +19,6 @@
                 ;;
 esac
 
-go build -o "${TARGET}" -tags pkcs11 --ldflags "${LDFLAGS}" ${GO_BUILDMODE} "${SOURCE}"
+${GO} build -o "${TARGET}" -tags pkcs11 --ldflags "${LDFLAGS}" ${GO_BUILDMODE} "${SOURCE}"
 
 ln -sf "$(basename "${TARGET}")" build/docker
