--- xen-4.1.2/config/StdGNU.mk.sav	2012-03-08 12:30:26.047303578 -0700
+++ xen-4.1.2/config/StdGNU.mk	2012-03-08 12:31:48.826696508 -0700
@@ -56,6 +56,9 @@
 XEN_CONFIG_DIR = $(CONFIG_DIR)/xen
 XEN_SCRIPT_DIR = $(XEN_CONFIG_DIR)/scripts
 
+# Allow xend to be disabled in lieu of only libxl
+CONFIG_ENABLE_XEND ?= y
+
 SOCKET_LIBS =
 CURSES_LIBS = -lncurses
 PTHREAD_LIBS = -lpthread
--- xen-4.1.2/tools/Makefile.sav	2012-03-08 11:25:10.177301492 -0700
+++ xen-4.1.2/tools/Makefile	2012-03-08 12:33:48.807302767 -0700
@@ -15,7 +15,7 @@
 SUBDIRS-$(CONFIG_X86) += firmware
 SUBDIRS-$(ACM_SECURITY) += security
 SUBDIRS-y += console
-SUBDIRS-y += xenmon
+SUBDIRS-$(CONFIG_ENABLE_XEND) += xenmon
 SUBDIRS-$(VTPM_TOOLS) += vtpm_manager
 SUBDIRS-$(VTPM_TOOLS) += vtpm
 SUBDIRS-y += xenstat
@@ -36,17 +36,19 @@
 
 SUBDIRS-y += xenpmd
 SUBDIRS-y += libxl
-SUBDIRS-y += remus
+SUBDIRS-$(CONFIG_ENABLE_XEND) += remus
 SUBDIRS-$(CONFIG_X86) += xenpaging
 SUBDIRS-$(CONFIG_X86) += debugger/gdbsx
 SUBDIRS-$(CONFIG_X86) += debugger/kdd
 
 # These don't cross-compile
 ifeq ($(XEN_COMPILE_ARCH),$(XEN_TARGET_ARCH))
+ifneq ($(XEN_ENABLE_XEND),y)
 SUBDIRS-$(PYTHON_TOOLS) += python
 SUBDIRS-$(PYTHON_TOOLS) += pygrub
 SUBDIRS-$(OCAML_TOOLS) += ocaml
 endif
+endif
 
 # For the sake of linking, set the sys-root
 ifneq ($(CROSS_COMPILE),)
--- xen-4.1.2/tools/misc/Makefile.sav	2012-03-08 12:23:10.107321955 -0700
+++ xen-4.1.2/tools/misc/Makefile	2012-03-08 12:34:54.466721729 -0700
@@ -19,11 +19,13 @@
 SUBDIRS-$(CONFIG_MINITERM) += miniterm
 SUBDIRS := $(SUBDIRS-y)
 
-INSTALL_BIN-y := xencons
+INSTALL_BIN-y :=
+INSTALL_BIN-$(CONFIG_ENABLE_XEND) += xencons
 INSTALL_BIN-$(CONFIG_X86) += xen-detect
 INSTALL_BIN := $(INSTALL_BIN-y)
 
-INSTALL_SBIN-y := xm xen-bugtool xen-python-path xend xenperf xsview xenpm xen-tmem-list-parse gtraceview gtracestat xenlockprof xenwatchdogd
+INSTALL_SBIN-y := xenperf xsview xenpm xen-tmem-list-parse gtraceview gtracestat xenlockprof xenwatchdogd
+INSTALL_SBIN-$(CONFIG_ENABLE_XEND) += xm xend xsview xen-python-path xen-bugtool
 INSTALL_SBIN-$(CONFIG_X86) += xen-hvmctx xen-hvmcrash
 INSTALL_SBIN-$(CONFIG_MIGRATE) += xen-hptool
 INSTALL_SBIN := $(INSTALL_SBIN-y)
--- xen-4.1.2/tools/hotplug/NetBSD/Makefile.sav	2012-03-08 12:54:56.247306596 -0700
+++ xen-4.1.2/tools/hotplug/NetBSD/Makefile	2012-03-08 12:57:50.337339410 -0700
@@ -8,7 +8,9 @@
 XEN_SCRIPTS += vif-ip
 
 XEN_SCRIPT_DATA =
-XEN_RCD_PROG = rc.d/xencommons rc.d/xend rc.d/xendomains rc.d/xen-watchdog
+XEN_RCD_PROG-y := rc.d/xencommons rc.d/xendomains rc.d/xen-watchdog
+XEN_RCD_PROG-$(CONFIG_ENABLE_XEND) += rc.d/xend 
+XEN_RCD_PROG := $(XEN_RCD_PROG-y)
 
 .PHONY: all
 all:
